project: private marketplace docs (#259)
mia-platform/documentation: docs/marketplace/add_to_marketplace/contributing_overview.md

sidebar_label: Add a new element to the Marketplace
---

In this section, we dive into the process of creating a new component for the Marketplace, analyzing the main differences between different types of components and providing some configuration examples.

> 

Reviewer: Let's add a note that marketplace items can only be created by people with access to the Mia-Platform Console CMS, therefore you may have to ask your Mia-Platform referent to add your marketplace items for you.

## Public and private marketplace

The Mia-Platform Marketplace is the place where every Company can find most of the components needed to set up its architecture. It is publicly available on the Console, meaning that all Companies present on the platform can view and make use of its components.

However, a Company might want to have its own private Marketplace, with private components that must not be shared with everyone else on the Console. 

To achieve this, a Company Owner or a Project Administrator can create new private components **from the CMS** by specifying their Company's identifier in the `tenantId` field.

Company Owners and Project Administrators can create, delete and update Company-wide markeplace elements, but they cannot manage public Marketplace items.  
The Console Super User role, instead, has permission to manage all Marketplace items from the CMS without any limitations.

> If you want to contribute to the **public Mia-Platform Marketplace** with a new component instead, you can [open an issue](https://github.com/mia-platform/community/issues/new?labels=marketplace&template=marketplace-contribution.md&title=Add+new+marketplace+item) on GitHub, wherein you will find what information is needed for the request.

Reviewer 2: ```suggestion
If you want to contribute to the **public Mia-Platform Marketplace** with a new component instead, and make it available for everyone, you can [open an issue](https://github.com/mia-platform/community/issues/new?labels=marketplace&template=marketplace-contribution.md&title=Add+new+marketplace+item) on GitHub, wherein you will find what information is needed for the request.
```

- **categoryId**: a label to help categorize components by their purpose or use case. As specified before, categories are only created internally at Mia-Platform. The `categoryId` of a component uniquely determines both a specific category and a sub-category (e.g. Start from Code (category) - Node.js (subcategory) will be identified by the `categoryId` "nodejs")
- **supportedBy**: a label to identify the company that has produced the component (only used if `supportedByImage` is not provided)
- **image** and **supportedByImage**: respectively the image that will be associated with the component and the image that will be associated with the company that has produced it.

> - **tenantId** (optional): the Company id, only needed if the component is a private Marketplace component that must not be generally available on the Mia-Platform Marketplace.

Reviewer: ```suggestion
- **tenantId** (optional): the Company id, only needed if the component is a private, Company-owned, Marketplace component that must not be generally available on the Mia-Platform Marketplace.
```

> - **tenantId** (optional): the Company id, only needed if the component is a private Marketplace component that must not be generally available on the Mia-Platform Marketplace.

Reviewer 2: ```suggestion
- **tenantId** (optional): the Company id, only needed if the component is a private Marketplace component for your Company and therefore must not be generally available on the public Mia-Platform Marketplace.
```

- **categoryId**: a label to help categorize components by their purpose or use case. As specified before, categories are only created internally at Mia-Platform. The `categoryId` of a component uniquely determines both a specific category and a sub-category (e.g. Start from Code (category) - Node.js (subcategory) will be identified by the `categoryId` "nodejs")
- **supportedBy**: a label to identify the company that has produced the component (only used if `supportedByImage` is not provided)
- **image** and **supportedByImage**: respectively the image that will be associated with the component and the image that will be associated with the company that has produced it.

> - **tenantId** (optional): the Company id, only needed if the component is a private, Company-owned, Marketplace component that must not be generally available on the Mia-Platform Marketplace.

Author: ```suggestion
- **tenantId** (optional): the Company id, only needed if the component is a private, Company-owned, Marketplace component that must not be generally available on the public Mia-Platform Marketplace.
```
